lib/matplotlib/legend.py

```diff
@@ -1145,6 +1142,12 @@ def _find_best_position(self, width, height, renderer, consider=None):
             candidates.append((badness, idx, (l, b)))
 
         _, _, (l, b) = min(candidates)
+
+        if self._loc_used_default and time.perf_counter() - start_time > 1:
+            cbook._warn_external(
+                'Creating legend with loc="best" can be slow with large '
```

Comment:
include the actual run time?  Wish we could use the walrus here ;)

Reply:
would rather not, as if we do I'm fairly worried the next thing is that someone will ask for the warning threshold to be configurable and whatnot.

Reply:
fair enough.